public/app/plugins/datasource/loki/LanguageProvider.ts

```diff
@@ -229,9 +251,11 @@ export default class LokiLanguageProvider extends LanguageProvider {
     options?: { streamSelector?: string; timeRange?: TimeRange }
   ): Promise<string[]> {
     const label = encodeURIComponent(this.datasource.interpolateString(labelName));
-    const streamParam = options?.streamSelector
-      ? encodeURIComponent(this.datasource.interpolateString(options.streamSelector))
-      : undefined;
+    // Loki don't allow empty streamSelector {}, so we should not send it.
+    const streamParam =
+      options?.streamSelector && options.streamSelector !== '{}'
```

Comment:
nit pick, I would extract `'{}'` to a constant (e.g. `const EMPTY_SELECTOR = '{}'`) and used that across codebase.

Reply:
fixed